src/stan/mcmc/hmc/xhmc/adapt_unit_e_xhmc.hpp

```diff
@@ -0,0 +1,43 @@
+#ifndef STAN_MCMC_HMC_xhmc_ADAPT_UNIT_E_XHMC_HPP
```

Comment:
This lower-case xhmc again.  But what I really wanted to ask is whether it would be possible to factor the mass matrix adaptation from the step size adaptation so that unit_e is really just a specified mass matrix with no adaptation of the mass matrix.  Or will that make the arithmetic slower to actually have the mass matrix in there doing nothing?